src/shader/gfx_webgl.js

```diff
@@ -0,0 +1,4716 @@
+                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+                gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+
+                var texUnit = 0;
+                if (s_geo.getIndexTexture()) {
+                    if (!sp.IG_indexTexture) {
+                        sp.IG_indexTexture = texUnit++;
+                    }
+                }
+                if (s_geo.getCoordinateTexture(0)) {
+                    if (!sp.IG_coordinateTexture) {
+                        sp.IG_coordinateTexture = texUnit++;
+                    }
+                }
+            }
+            else if ((s_gl.binaryGeometry != 0 || s_gl.externalGeometry != 0) && s_geo._vf["idsPerVertex"] == true) { //MultiPart
+                var shader = s_app._shader;
+                if(shader && x3dom.isa(s_app._shader, x3dom.nodeTypes.CommonSurfaceShader)) {
+                    if (shader.getMultiVisibilityMap()) {
+                        sp.multiVisibilityMap = 0;
+                        var visTex = x3dom.Utils.findTextureByName(s_gl.texture, "multiVisibilityMap");
+                        sp.multiVisibilityWidth = visTex.texture.width;
+                        sp.multiVisibilityHeight = visTex.texture.height;
+                        gl.activeTexture(gl.TEXTURE0);
+                        gl.bindTexture(gl.TEXTURE_2D, visTex.texture);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+                        gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+                    }
+                }
+            }
+
+            if (shape.isSolid()) {
+                this.stateManager.enable(gl.CULL_FACE);
+
+                if (shape.isCCW()) {
+                    this.stateManager.frontFace(gl.CCW);
+                }
+                else {
+                    this.stateManager.frontFace(gl.CW);
+                }
+            }
+            else {
+                this.stateManager.disable(gl.CULL_FACE);
+            }
+
+            //===========================================================================
+            // Set DepthMode
+            //===========================================================================
+            var depthMode = s_app ? s_app._cf.depthMode.node : null;
+            if (depthMode)
+            {
+                if (depthMode._vf.enableDepthTest)
+                {
+                    //Enable Depth Test
+                    this.stateManager.enable(gl.DEPTH_TEST);
+
+                    //Set Depth Mask
+                    this.stateManager.depthMask(!depthMod
```

Comment:
why in src/shader ? Probably the file should just stay in src/ .
z is not used, so ok.